keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -0,0 +1,1009 @@
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The proposed enhancement adds a new `VolumeSource` to Kubernetes that supports OCI images and/or OCI artifacts.
+This allows users to package files and share them among containers in a pod without including them in the main image,
+thereby reducing vulnerabilities and simplifying image creation.
+
+While OCI images are well-supported by Kubernetes and CRI, extending support to
+OCI artifacts involves recognizing additional media types within container
+runtimes, implementing custom lifecycle management, resolution of artifact
+registry referrers use pattern for artifacts, and ensuring appropriate
+validation and security measures.
+
+## Motivation
+
+Supporting OCI images and artifacts directly as a `VolumeSource` allows
+Kubernetes to focus on OCI standards as well as allows to store and distribute
+any content using OCI registries. This allows the project to grow into use cases
+which go beyond running particular images.
+
+### Goals
+
+- Introduce a new `VolumeSource` type that allows mounting OCI images and/or artifacts.
+- Simplify the process of sharing files among containers in a pod.
+
+### Non-Goals
+
+- This proposal does not aim to replace existing `VolumeSource` types.
+- This proposal does not address other use cases for OCI objects beyond file sharing among containers in a pod.
+
+## Proposal
+
+We propose to add a new `VolumeSource` that supports OCI images and/or artifacts. This `VolumeSource` will allow users to mount an OCI object
+directly into a pod, making the files within the image accessible to the containers without the need to include them in the main image and to be able to host them in OCI compatible registries.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a Kubernetes user, I want to share a configuration file among multiple containers in a pod without including the file in my main image, so that I can
+minimize security risks and image size. 
+
+Beside that, I want:
+- to package this file in an OCI object to take advantage of OCI distribution.
+- the image to be downloaded with the same credentials that kubelet using for other images.
+- to be able to use image pull secrets when downloading the image if an image is from the registry that requires image pull secrets.
```

Comment:
What none of these stories specifically clarify or not is what their lifecycle needs are - i.e. do we need to support changing the volume image source and have it update?  What happens with pull policy?

I would expect either a story that talks about whether updates of images is needed, and if not I think we should call it out in non-goals.

Reply:
I added this to the first story by using a moving tag like `latest`.